File: invision-mobile/lib/src/core/enums/campaign_type.dart
enum CampaignType {
  promotion,
  discount,
  sampling,
  display,
  posm,
  buyGetFree;

  String get label => switch (this) {
    promotion => 'Promotion',
    discount => 'Discount',
    sampling => 'Sampling',
    display => 'Display',
    posm => 'POSM',
    buyGetFree => 'Buy & Get Free',
  };

  static CampaignType fromString(String value) {
    return switch (value) {
      'promotion' => CampaignType.promotion,
      'discount' => CampaignType.discount,
      'sampling' => CampaignType.sampling,
      'display' => CampaignType.display,
      'posm' => CampaignType.posm,
      'buy_get_free' => CampaignType.buyGetFree,
      _ => CampaignType.promotion,
    };
  }
}

File: invision-backend/app/Enums/CampaignType.php
<?php

namespace App\Enums;

enum CampaignType: string
{
    case Promotion = 'promotion';
    case Discount = 'discount';
    case Sampling = 'sampling';
    case Display = 'display';
    case Posm = 'posm';
    case BuyGetFree = 'buy_get_free';

    public function label(): string
    {
        return match ($this) {
            self::Promotion => 'Promotion',
            self::Discount => 'Discount',
            self::Sampling => 'Sampling',
            self::Display => 'Display',
            self::Posm => 'POSM',
            self::BuyGetFree => 'Buy & Get Free',
        };
    }
}
